checks/evaluation/permissions/permissions.go

```diff
@@ -99,77 +143,215 @@ func applyScorePolicy(results *checker.TokenPermissionsData, c *checker.CheckReq
 				loc.Snippet = newStr(r.File.Snippet)
 			}
 		}
-
 		text, err := createText(r)
 		if err != nil {
-			return checker.MinResultScore, err
+			return findings, err
 		}
 
-		msg, err := createLogMsg(r.LocationType)
+		f, err := createFinding(r.LocationType, text, loc)
 		if err != nil {
-			return checker.InconclusiveResultScore, err
+			return findings, err
 		}
-		msg.Finding = msg.Finding.WithMessage(text).WithLocation(loc)
+
 		switch r.Type {
-		case checker.PermissionLevelNone, checker.PermissionLevelRead:
-			msg.Finding = msg.Finding.WithOutcome(finding.OutcomePositive)
-			dl.Info(msg)
-		case checker.PermissionLevelUnknown:
-			dl.Debug(msg)
+		case checker.PermissionLevelNone:
+			f = f.WithOutcome(finding.OutcomePositive)
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelNone),
+			})
+		case checker.PermissionLevelRead:
+			f = f.WithOutcome(finding.OutcomePositive)
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelRead),
+			})
 
+		case checker.PermissionLevelUnknown:
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelUnknown),
+			})
 		case checker.PermissionLevelUndeclared:
+			var locationType LocationType
+			//nolint:gocritic
 			if r.LocationType == nil {
-				return checker.InconclusiveResultScore,
-					sce.WithMessage(sce.ErrScorecardInternal, "locationType is nil")
+				locationType = PermissionLocationNil
+			} else if *r.LocationType == checker.PermissionLocationTop {
+				locationType = PermissionLocationTop
+			} else {
+				locationType = PermissionLocationNotDeclared
+			}
+			permissionType := permTypeToEnum(r.Name)
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelUndeclared),
+				"LocationType":    int(locationType),
+				"PermissionType":  int(permissionType),
+			})
+		case checker.PermissionLevelWrite:
+			var locationType LocationType
+			switch *r.LocationType {
+			case checker.PermissionLocationTop:
+				locationType = PermissionLocationTop
+			case checker.PermissionLocationJob:
+				locationType = PermissionLocationJob
+			default:
+				locationType = PermissionLocationNotDeclared
 			}
+			permissionType := permTypeToEnum(r.Name)
+			f = f.WithValues(map[string]int{
+				"PermissionLevel": int(PermissionLevelWrite),
+				"LocationType":    int(locationType),
+				"PermissionType":  int(permissionType),
+			})
+			f = f.WithOutcome(finding.OutcomeNegative)
```

Comment:
hmm, I don't think all write permissions are considered bad under our current scoring policy, just top level ones. but this seems to match what the probe definition for job level currently seems to say.